### ISSUE TITLE
Have installed everything as per directions.  Game still launches without EE.  

### ISSUE THREAD

**Issue by BiggestRonald** [elishacloud/Silent-Hill-2-Enhancements#541]
Hello.  Recently, I installed the "Fully Extracted Version" of SH2 from MyAbandonware.  After, I installed as per instruction, putting the EE files in game directory. However, I am only able to open the game through the directory (the Enhanced Executable) In honestly, I'm new to Windows as well, so this could be a simple mistake.  However, I was told I should be able to launch the game from my Start Menu.  The game does not show up as an application there which is troubling and likely part of the problem.  Of course, I don't know what the problem is.

Below I linked an image of my directory, or at least what I think a directory is.  If you have any tips, please let me know.  Can't wait to play this game.  If any more information is needed, please let me know. 

![Screen Shot 2022-06-30 at 11 44 17 PM](https://user-images.githubusercontent.com/108505972/176819345-21b71d39-7d02-4ca1-90ed-e77654840b03.png)

**Comment by Polymega** [comment]
Hi there,

Sorry you're having issues. Looking at you directory, it seems a handful of files are missing. For example, I see you have downloaded the DSOAL package but are missing the alsoft.ini file from it. Although, the files you have missing shouldn't prevent the project from running, from what I can see.

Your directory should look like this, depending on what packages you downloaded for our project:
http://enhanced.townofsilenthill.com/SH2/troubleshoot.htm#directoryTree

Now, once you launch the game the game itself will also create some files in the directory, which I don't see. So to confirm: Is the issue you're having that the game won't launch with our project or that you cannot get the game to launch at all?

> However, I was told I should be able to launch the game from my Start Menu. The game does not show up as an application there which is troubling and likely part of the problem.

This isn't a problem and is normal. You technically didn't install the game so there are no registries written to your Windows by the game (which would include a Start menu shortcut). Instead of installing the base game, you grabbed all of the extracted games files _from_ an install.

In other words, you got the game without actually having to install it. If you want, you can right-click sh2pc.exe and create a shortcut then place that shortcut wherever you want to launch the game through that.

**Comment by AeroWidescreen** [comment]
You can create a shortcut on your desktop by right clicking sh2pc.exe and selecting: Send to -> Desktop (shortcut). Or you could drag and drop sh2pc.exe to your taskbar (the big bar at the bottom) to create a shortcut there instead.

**Comment by BiggestRonald** [comment]
@Polymega Thank you for your speedy response and for the guide regarding how my folder should look.  Below is what the directory looks like when the game is launched.  My issue is not launching the game but rather, none of the features, graphical enhancements etc are actually in the game, when I launch the game.  This includes menus, cutscenes, audio, gameplay and all.  So again, I am launching the sh2pc.exe file from the folder/directory, but none of the EE features apply in any way.  



![Screen Shot 2022-07-01 at 1 17 08 AM](https://user-images.githubusercontent.com/108505972/176828642-efa6bf13-3cf2-4e6d-b383-ea894e0b4a92.png)

**Comment by Polymega** [comment]
Where is this game folder located exactly? Can you share its filepath?

**Comment by Polymega** [comment]
Hi again,

[Can you please update the project](https://www.youtube.com/watch?v=LylafxQta9w) then open the Configuration Tool and click the "Default" button to see if that fixes things? We've just had another ticket opened where, at first, they couldn't launch the game, and then they could be the EE project wasn't taking effect.

Restoring settings to "default" seems to have fixed it for them: 
https://github.com/elishacloud/Silent-Hill-2-Enhancements/issues/542#issuecomment-1172715033

**Comment by BiggestRonald** [comment]
@Polymega Think I may have found an additional problem, I never installed the Configuration Tool or the setup tool.  Can you link those?  From there I'll retry everything.

**Comment by Polymega** [comment]
No worries. Both of these tools have are new since you first installed the project. The Setup Tool can be found here: http://enhanced.townofsilenthill.com/SH2/install.htm#usingEE

[Note for people who have been previously using the project.
](https://youtu.be/LylafxQta9w?t=160)
Once you download/update the files the Configuration Tool can be found in the game's menu.

However, for your particular issue, do me a favor: Once you update the project (through the use of the Setup Tool) [download this package](https://github.com/elishacloud/Silent-Hill-2-Enhancements/files/9031801/d3d8-4.zip) and replace files in the game's main directory.

[This package is "d3d8-4.zip" found from this troubleshooting thread.](https://github.com/elishacloud/Silent-Hill-2-Enhancements/issues/542#issuecomment-1172737224) (This person had similar issues as you.)

**Comment by BiggestRonald** [comment]
@Polymega   Thank you for your help thus far but, new issue.  After installing the Setup file and attempting to launch I was met with this message: 


dlll:<utf8>C:\Users\Ron\AppData\Local\Temp\is-P6JIN.tmp\BytesToString.dll.  

I took it up with Microsoft support for 4 hours, but they had no clue what it was.  Once, I figure this last issue out I should be able to play.  The setup worked earlier today but upon reinstalment (for unrelated reasons) I was no longer able to access it and simply met with the code above.

**Comment by Polymega** [comment]
If you right click SH2EEsetup.exe and go to Properties then the Details tab what version number does it say?

**Comment by nipkownix** [comment]
@BiggestRonald you might be missing the C++ Redistributes. Please try installing this: 

https://aka.ms/vs/17/release/vc_redist.x86.exe

**Comment by BiggestRonald** [comment]
@Polymega 

File Version: 1.0.4.0

Product version: 1.0.4

**Comment by BiggestRonald** [comment]
@Polymega @nipkownix  I was missing the C ++ Redistributes as believed.  Thank you both for your help!

**Comment by Polymega** [comment]
Thanks, nipkow!

**Comment by elishacloud** [comment]
> you might be missing the C++ Redistributes. Please try installing this:
> 
> https://aka.ms/vs/17/release/vc_redist.x86.exe

@nipkownix, it would be great if you could statically link the dlls needed from this distributable, [like we do with the enhancement module](https://github.com/elishacloud/Silent-Hill-2-Enhancements/blob/0345ca0aa48caa80c9eb914d0cd52de5ec59517e/sh2-enhce.vcxproj#L273), so that users don't need to download or install this distributable.

**Comment by nipkownix** [comment]
Hmm, thanks for the heads up. Gotta admit I didn't know this was possible.
I recently had to reinstall Windows, though, so I no longer have the older VS with XP build tools.
If you could [make the change](https://github.com/nipkownix/SH2EE-web-installer/tree/master/includes/BytesToString/src) and send me a compiled dll, that would be nice.
Otherwise, I'll look into it myself in a couple of weeks or so.

**Comment by elishacloud** [comment]
> If you could [make the change](https://github.com/nipkownix/SH2EE-web-installer/tree/master/includes/BytesToString/src) and send me a compiled dll, that would be nice.

Here is the updated build:  [BytesToString.zip](https://github.com/elishacloud/Silent-Hill-2-Enhancements/files/9049193/BytesToString.zip)

**Comment by nipkownix** [comment]
Awesome! Many thanks. I'll soon send Polymega an update.
